include/unifex/unstoppable_baton.hpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2019-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/receiver_concepts.hpp>
+
+#include <atomic>
+#include <type_traits>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex {
+
+namespace _ub {
+
+struct _op_base;
+
+template <typename Receiver>
+struct _operation {
+  struct type;
+};
+
+template <typename Receiver>
+using operation = typename _operation<Receiver>::type;
+
+struct unstoppable_baton;
+
+struct _sender {
+  template <template <class...> class Variant,
+            template <class...> class Tuple>
+  using value_types = Variant<Tuple<>>;
+
+  template <template <class...> class Variant>
+  using error_types = Variant<std::exception_ptr>;
+
+  static constexpr bool sends_done = false;
+
+  explicit _sender(unstoppable_baton& baton) noexcept
+    : baton_(&baton) {}
+
+  template (typename Receiver)
+    (requires receiver_of<Receiver>)
+  operation<std::decay_t<Receiver>> connect(Receiver&& r) const noexcept(
+      std::is_nothrow_constructible_v<std::decay_t<Receiver>, Receiver&&>) {
```

Comment:
The `&&` here is superfluous. Either `Receiver` is an lvalue reference type, in which case the `&&` collapses away, or else it isn't reference qualified, in which case `Receiver` and `Receiver&&` are semantically indistinguishable.

```suggestion
      std::is_nothrow_constructible_v<std::decay_t<Receiver>, Receiver>) {
```